### PR TITLE
Minifier-friendly references to properties

### PR DESCRIPTION
**Pull request by sd-yip**
**Description of the change**
Dynamically matching property names in a switch is hard to detect and optimize. Currently, it breaks minifier outputs under aggressive minification settings. This PR modifies the corresponding part for easier usage with a minifier.

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation in the README and/or documentation directory
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: src/React.js [hunk 1]
```diff
@@ -1,40 +1,14 @@
...
-
-      case "unsafeComponentWillUpdate":
-        instance["UNSAFE_componentWillUpdate"] = function (a, b) { return value(a)(b)(); };
-        break;
-
-      default:
-        throw new Error("[purescript-react] Not a component property: " + prop);
+  function curry2(f) {
+    return f === undefined ? f : function (a, b) {
+      return f(a)(b)
```
Comment (natefaubion) [inline src/React.js:6]: I don't think these are correct. They are missing the `()` for invoking the effects (ie, it's not just a curry).

Comment (sd-yip) [reply src/React.js]: I see. Previously, I applied the amount of arguments + 1 as the currying number for `UNSAFE_componentWillReceiveProps` and `UNSAFE_componentWillUpdate`, but not sure why I have missed them for the remaining.

Now the invocation and currying should be fixed. Please take another look.

**Review by natefaubion — APPROVED** [review]

**Comment by JordanMartinez** [comment]
:ping_pong: @thomashoneyman

**Review by thomashoneyman — APPROVED** [review]

**Comment by thomashoneyman** [comment]
https://github.com/purescript-contrib/purescript-react/releases/tag/v10.0.1

**Pull request merged**